include/libopencm3/sam/common/pmc_common_all.h

```diff
@@ -27,160 +34,223 @@
 
 /* --- Power Management Controller (PMC) registers ----------------------- */
 
-/* System Clock Enable Register */
+/** System Clock Enable Register */
 #define PMC_SCER			MMIO32(PMC_BASE + 0x0000)
 
-/* System Clock Disable Register */
+/** System Clock Disable Register */
 #define PMC_SCDR			MMIO32(PMC_BASE + 0x0004)
 
-/* System Clock Status Register */
+/** System Clock Status Register */
 #define PMC_SCSR			MMIO32(PMC_BASE + 0x0008)
 
-/* Main Oscillator Register */
+/** Main Oscillator Register */
 #define CKGR_MOR			MMIO32(PMC_BASE + 0x0020)
 
-/* Main Clock Frequency Register */
+/** Main Clock Frequency Register */
 #define CKGR_MCFR			MMIO32(PMC_BASE + 0x0024)
 
-/* PLLA Register */
+/** PLLA Register */
 #define CKGR_PLLAR			MMIO32(PMC_BASE + 0x0028)
 
-/* Master Clock Register */
+/** Master Clock Register */
 #define PMC_MCKR			MMIO32(PMC_BASE + 0x0030)
 
-/* Programmable Clock 0 Register */
+/** Programmable Clock 0 Register */
 #define PMC_PCK0			MMIO32(PMC_BASE + 0x0040)
 
-/* Programmable Clock 1 Register */
+/** Programmable Clock 1 Register */
 #define PMC_PCK1			MMIO32(PMC_BASE + 0x0044)
 
-/* Programmable Clock 2 Register */
+/** Programmable Clock 2 Register */
 #define PMC_PCK2			MMIO32(PMC_BASE + 0x0048)
 
-/* Interrupt Enable Register */
+/** Interrupt Enable Register */
 #define PMC_IER				MMIO32(PMC_BASE + 0x0060)
 
-/* Interrupt Disable Register */
+/** Interrupt Disable Register */
 #define PMC_IDR				MMIO32(PMC_BASE + 0x0064)
 
-/* Status Register */
+/** Status Register */
 #define PMC_SR				MMIO32(PMC_BASE + 0x0068)
 
-/* Interrupt Mask Register */
+/** Interrupt Mask Register */
 #define PMC_IMR				MMIO32(PMC_BASE + 0x006C)
 
-/* Fast Startup Mode Register */
+/** Fast Startup Mode Register */
 #define PMC_FSMR			MMIO32(PMC_BASE + 0x0070)
 
-/* Fast Startup Polarity Register */
+/** Fast Startup Polarity Register */
 #define PMC_FSPR			MMIO32(PMC_BASE + 0x0074)
 
-/* Fault Output Clear Register */
+/** Fault Output Clear Register */
 #define PMC_FOCR			MMIO32(PMC_BASE + 0x0078)
 
-/* Write Protect Mode Register */
+/** Write Protect Mode Register */
 #define PMC_WPMR			MMIO32(PMC_BASE + 0x00E4)
 
-/* Write Protect Status Register */
+/** Write Protect Status Register */
 #define PMC_WPSR			MMIO32(PMC_BASE + 0x00E8)
 
 
 /* --- Register contents --------------------------------------------------- */
 
+/**
+ * @addtogroup pmc_scer PMC System Clock Enable Register (PMC_SCER)
+ * @{
+ */
 
-/* --- PMC System Clock Enable Register (PMC_SCER) ------------------------- */
-
-/* Programmable Clock Output Enable */
```

Comment:
Here I decided to make groups for contents of registers.
A similar method is used here (but not necessarily grouped by register, I don’t know): https://github.com/libopencm3/libopencm3/blob/467522778329d6f41781a6c951b77d6ff6744de6/include/libopencm3/nrf/common/i2c.h#L75-L82

So far, this method of grouping makes sense here.
Register contents for the same register may be defined in multiple `pmc_*_common.h` files, and with this grouping they end up in one place.
(Without this method of grouping, they would also end up in one place, in the main page of the PMC module (which is a large place).)

Reply:
(Answered as of https://github.com/libopencm3/libopencm3/pull/1411#discussion_r917442391)